### PR TITLE
Hide cookie banner if cross-origin message has hideCookieBanner set to true

### PR DESCRIPTION
**Pull request by alex-ju**
## What
Update cookie banner component to hide if cross-origin message has `hideCookieBanner` set to `true`.

The cookie banner script listens for cross-origin messages and if they're coming from a `publishing.service.gov.uk` domain and have `hideCookieBanner` set to `true` it hides the banner. 

## Why
This behaviour is required for [previewing GOV.UK pages from draft stack in Content Publisher without obstructing the view](https://github.com/alphagov/content-publisher/pull/1240).

## Visual Changes
No visual changes.

[Trello card](https://trello.com/c/QXbQKBG3)

### CODE REVIEW

File: app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js [hunk 1]
```diff
@@ -11,12 +11,16 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     this.$module.setCookieConsent = this.setCookieConsent.bind(this)
 
     this.$module.cookieBanner = document.querySelector('.gem-c-cookie-banner')
-    this.$module.cookieBannerConfirmationMessage = document.querySelector('.gem-c-cookie-banner__confirmation')
+    this.$module.cookieBannerConfirmationMessage = this.$module.querySelector('.gem-c-cookie-banner__confirmation')
 
     // Temporary check while we have 2 banners co-existing.
```
Comment (benthorner) [inline app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js]: Not really part of this PR, but I'm guessing this doesn't apply anymore?

File: app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js [hunk 2]
```diff
@@ -84,5 +88,41 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
...
+    var trustedDomain = 'publishing.service.gov.uk'
+    var origin = event.origin
+
+    // Return if no origin is given or the browser doesn't support lastIndexOf
+    if (!origin || !origin.lastIndexOf) {
+      return
+    }
+
+    // Polyfill origin.endsWith(trustedDomain) for IE
+    var offset = trustedDomain && this ? origin.length - trustedDomain.length : -1
```
Comment (benthorner) [inline app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js]: Pretty sure we don't need to test `trustedDomain` exists here.

File: app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js (hunk 1 above)
Comment (alex-ju) [reply app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js]: Yep, I'll try and do some clean up on those comments.

File: app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js (hunk 2 above)
Comment (alex-ju) [reply app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js]: Good spot; leftover from refactoring
